modules/migrations/gitea_uploader.go

```diff
@@ -87,19 +87,22 @@ func (g *GiteaLocalUploader) MaxBatchInsertSize(tp string) int {
 }
 
 func fullURL(opts base.MigrateOptions, remoteAddr string) (string, error) {
-	var fullRemoteAddr = remoteAddr
 	if len(opts.AuthToken) > 0 || len(opts.AuthUsername) > 0 {
 		u, err := url.Parse(remoteAddr)
 		if err != nil {
 			return "", err
 		}
 		u.User = url.UserPassword(opts.AuthUsername, opts.AuthPassword)
 		if len(opts.AuthToken) > 0 {
-			u.User = url.UserPassword("oauth2", opts.AuthToken)
+			if opts.GitServiceType == structs.GogsService {
```

Comment:
we may should consider this "create git clone url" into the download interface ....